concourse/steps/release.py

```diff
@@ -390,17 +402,49 @@ def validate(self):
     def apply(
         self,
     ):
+        # clean repository if required
+        worktree_dirty = bool(self.github_helper._changed_file_paths())
+        if worktree_dirty:
+            self.github_helper._stash_changes()
+
+        # update version file
         next_cycle_dev_version = self._calculate_next_cycle_dev_version(
             release_version=self.release_version,
             version_operation=self.version_operation,
             prerelease_suffix=self.prerelease_suffix,
         )
-        # Prepare version file for next dev cycle
-        self.github_helper.create_or_update_file(
-            file_path=self.repository_version_file_path,
-            file_contents=next_cycle_dev_version,
-            commit_message=f'Prepare next dev cycle {next_cycle_dev_version}'
-        )
+
+        # update version file
+        with open(self.repository_version_file_path, 'w') as f:
+            f.write(next_cycle_dev_version)
+
+        try:
+            # call optional dev cycle commit callback
+            if self.dev_cycle_commit_callback:
+                self._invoke_dev_cycle_commit_callback(
+                    dev_cycle_commit_callback=self.dev_cycle_commit_callback,
+                    repo_dir=self.repo_dir,
+                    release_version=next_cycle_dev_version,
+                )
+
+            dev_cycle_commit = self.github_helper.index_to_commit(
```

Comment:
almost.. I admit, our API is a bit confusing there. There are two helpers: `GitHubRepositoryHelper` (from github/util.py) and GitHelper (from gitutil.py). The `index_to_commit` method is exposed from `GitHelper`. So you will also need to instantiate a `GitHelper` instance and use it.

nit: also, maybe it would be adequate to mv the callback-invoking into a separate (private) method

Reply:
I implemented a method `_invoke_callback` that has the required characteristics.
I also now make use of the `git_helper` in addition.